added support to autoSelectFirstItem of the autocomplete editor

leif-freshx opened the pull request:
This change resolves the feature request: [https://github.com/olifolkerd/tabulator/issues/2906](url)

I apologize if I have not followed some best practice in the GitHub world (this is my first GitHub pull request).

Should the documentation also be updated connected to the change (since a new property of the editorParams is added)?

Best regards,

Leif

leif-freshx commented on src/js/modules/edit.js:
```diff
@@ -1631,9 +1632,7 @@ Edit.prototype.editors = {
 					if(intialLoad && item.value == initialValue){
-						input.value = item.title;
-						item.element.classList.add("active");
-						current = true;
+						autoSelectIndex = idx;
 					}
```
The case for initialValue did not actually work with keyUp/keyDown prior to the addition of editorParams.autoSelectFirstItem but I stumbled on that bug while testing my changes.

The issue was that if the initial value was an item somewhere in the list and the user pressed down arrow, it actually selected the first item and ignored the position of the initial value (but that was fixed by this fix)

leif-freshx commented on src/js/modules/edit.js:
```diff
@@ -1601,7 +1601,7 @@ Edit.prototype.editors = {
 		function fillList(items, intialLoad){
 			var current = false;
-			var autoSelectIndex = -1;
+			var autoSelectIndex = editorParams.autoSelectFirstItem ? 0 : -1;
```
Made a slightly more elegant solution that combines the addition and the bugfix (sorry for the multiple pushes)

olifolkerd commented:
Hey @leif-freshx 

Thanks for submitting the PR,

I will look to include it in the 4.9 release.

Cheers

Oli :)

olifolkerd commented:
Hey @leif-freshx 

Following a review of the autocomplete editor in preparation for the 5.0 release i have decided to look at rebuilding the editor and integrate this functionality into the existing ***defaultValue*** property.

Thanks for the idea though, i would love to receive other PR's in future.

Cheers

Oli :)

leif-freshx commented:
Hi Oli,

Please note that the feature I added is completely different from
defaultValue. It helps when you are searching for data using e.g. ajax or
other logic to search for data based on what the user is typing. When the
user has typed enough to find the item, they can click enter with the
feature I added but it would be impossible when defining the column to know
what the default value is...

Default value only works if you know up front which items exist in the
suggestions list which you don't do in many cases...

/Leif

*______________________________________________________*

*Leif Lundberg*

+46 (0)70 357 08 09 | leif@freshx.se | lundberg@qvanti.se

Customer Support: +46 (0)10 884 84 08 | support@qvanti.se





Den sön 22 nov. 2020 kl 17:02 skrev Oli Folkerd <notifications@github.com>:

> Hey @leif-freshx <https://github.com/leif-freshx>
>
> Following a review of the autocomplete editor in preparation for the 5.0
> release i have decided to look at rebuilding the editor and integrate this
> functionality into the existing *defaultValue* property.
>
> Thanks for the idea though, i would love to receive other PR's in future.
>
> Cheers
>
> Oli :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/olifolkerd/tabulator/pull/3034#issuecomment-731769527>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APZ5GQBIRNGCPJ3CH6B72CTSREY2BANCNFSM4SSBX3YA>
> .
>
